File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/BannerDemoActivity.java
package io.ganguo.demo.view.activity;

import android.content.Context;
import android.content.Intent;

import io.ganguo.demo.view.activity.base.GGHFRVModelActivity;
import io.ganguo.demo.view.activity.base.GGHFSRVModelActivity;
import io.ganguo.demo.viewmodel.activity.BannerDemoVModel;
import io.ganguo.vmodel.ViewModelActivity;
import io.ganguo.viewmodel.databinding.IncludeHfRecyclerBinding;

/**
 * Created by leo on 2017/4/28.
 * Banner - demo
 */
public class BannerDemoActivity extends GGHFRVModelActivity<IncludeHfRecyclerBinding, BannerDemoVModel> {

    public static Intent intentFor(Context context) {
        return new Intent(context, BannerDemoActivity.class);
    }

    @Override
    public BannerDemoVModel createViewModel() {
        return new BannerDemoVModel();
    }

    @Override
    public void onViewAttached(BannerDemoVModel viewModel) {
        super.onViewAttached(viewModel);
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/BannerDemoActivity.kt
package io.ganguo.demo.view.activity

import android.content.Context
import android.content.Intent

import io.ganguo.demo.view.activity.base.GGHFRVModelActivity
import io.ganguo.demo.viewmodel.activity.BannerDemoVModel
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.databinding.IncludeHfRecyclerBinding

/**
 * Created by leo on 2017/4/28.
 * Banner - demo
 */
class BannerDemoActivity : GGHFRVModelActivity<IncludeHfRecyclerBinding, BannerDemoVModel>() {

    override fun createViewModel(): BannerDemoVModel {
        return BannerDemoVModel()
    }


    companion object {

        fun intentFor(context: Context): Intent {
            return Intent(context, BannerDemoActivity::class.java)
        }
    }
}
